src/store/ducks/generic/plans.ts

```diff
@@ -172,3 +174,35 @@ export function getIRSPlansArray(
   const result = values((state as any)[reducerName].IRSPlansById);
   return result.filter((e: IRSPlan) => e.plan_intervention_type === interventionType);
 }
+
+/** RESELECT USAGE STARTS HERE */
+
+/** This interface represents the structure of IRS plan filter options/params */
+export interface IRSPlanFilters {
+  plan_title?: string /** IRS plan title */;
+}
+
+/** IRSPlansArrayBaseSelector select an array of all plans
+ * @param state - the redux store
+ */
+export const IRSPlansArrayBaseSelector = (planKey?: string) => (state: Registry): IRSPlan[] =>
+  values((state as any)[reducerName][planKey ? planKey : 'IRSPlansById']);
```

Comment:
Do we still need to pass planKey as an argument here. It seems to me that it is not needed in this module. The reason it is required in `src/store/plans` ducks, is because the store state in that ducks has 2 top level properties, and thus one has to choose which of them they will use when creating the selector.

Reply:
@p-netm I see what you mean. But can we leave it for scalability?

Reply:
Or we are sure multiple level properties will not occur in future for IRSPlans?

Reply:
> @p-netm I see what you mean. But can we leave it for scalability?

ok, cool